plugins/amqp/src/plugin.cpp

```diff
@@ -0,0 +1,748 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/chunk.hpp>
+#include <tenzir/concept/parseable/tenzir/kvp.hpp>
+#include <tenzir/config.hpp>
+#include <tenzir/data.hpp>
+#include <tenzir/plugin.hpp>
+
+#include <caf/expected.hpp>
+
+#if TENZIR_MACOS
+#  include <rabbitmq-c/amqp.h>
+#  include <rabbitmq-c/ssl_socket.h>
+#  include <rabbitmq-c/tcp_socket.h>
+#else
+#  include <amqp.h>
+#  include <amqp_ssl_socket.h>
+#  include <amqp_tcp_socket.h>
```

Comment:
Debian at least ships these headers as 
```
/usr/include/amqp.h
/usr/include/amqp_framing.h
/usr/include/amqp_ssl_socket.h
/usr/include/amqp_tcp_socket.h
```
(see https://packages.debian.org/bookworm/amd64/librabbitmq-dev/filelist) at least for the version 0.11.0 that they ship. I would be happy if some degree of backwards compatibility could be provided 🙏🏻

Reply:
> Maybe this macro directive needs to be version-based instead of OS-based?

I agree. According to rabbitmq-c's changelog, the change happened in 0.12.0:

> [rabbitmq-c v0.12.0](https://github.com/alanxz/rabbitmq-c/releases/tag/v0.12.0)
Changed
> * ...
> * Public headers have moved to rabbitmq-c/ directory
> * ...

Reply:
Are there any action items then, @Dakostu @dominiklohmann?

Reply:
Just use something like this:

```cpp
#if __has_include(<rabbitmq-c/amqp.h>)
#  include <rabbitmq-c/amqp.h>
#else
#  include <amqp.h>
#endif
```

Reply:
Let's see if b5e7c371e7 worked.

Reply:
Looks like it did!